.github/workflows/build-test-deploy.yml

```diff
@@ -137,906 +62,417 @@ jobs:
-          echo "narinfo-cache-negative-ttl = 0" >> /etc/nix/nix.conf
-      - uses: cachix/cachix-action@f5f67badd061acb62b5c6e25e763572ca8317004
-        with:
-          skipPush: true
-          installCommand: |
-            nix-channel --add ${{ env.NIX_NIXPKGS_CHANNEL }} nixpkgs
-            nix-channel --update
-            nix-env -iA nixpkgs.cachix
-          authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
-          name: composable-community
-      - run: |
-          nix --version
-          nix show-config
-          nix flake show --allow-import-from-derivation  --show-trace --fallback --debug --print-build-logs --keep-failed
-
-  cargo-fmt-check:
-    name: "Cargo fmt check"
+          chmod +x ./.github/workflows/nix-flake-check.sh
+          source ./.github/workflows/nix-flake-check.sh
+  
+  build-packages:
+    name: Build packages
+    strategy:
+      matrix:
+        arch: [x64-monster, arm]
     needs: 
-      - check-privilege
+      - privilege-check
+      - lfs-check
     runs-on:
       - self-hosted
       - linux
-      - x64
+      - ${{ matrix.arch }}
       - sre
     concurrency:
-      group: ${{ github.workflow }}-cargo-fmt-check-${{ github.event.pull_request.title }}
+      group: ${{ github.workflow }}-all-ci-packages-${{ matrix.arch }}-${{ github.event.pull_request.title }}
       cancel-in-progress: true
-    container:
-      image: niteo/nixpkgs-nixos-22.05:316b762afdb9e142a803f29c49a88b4a47db80ee
     steps:
+      # This isn't templated, since checkout needs to happen before templating occurs.
+      - name: Set up Nix
+        uses: cachix/install-nix-action@daddc62a2e67d1decb56e028c9fa68344b9b7c2a # v18
+        with:
+          install_url: https://releases.nixos.org/nix/${{ env.NIX_VERSION }}/install
+          nix_path: nixpkgs=channel:${{ env.NIXPKGS_CHANNEL }}
+          extra_nix_config: |
+            sandbox = relaxed
+            narinfo-cache-negative-ttl = 0      
+            system-features = kvm
+            http2 = false
+      - name: Set up Cachix
+        uses: cachix/cachix-action@6a9a34cdd93d0ae4b4b59fd678660efb08109f2f # v12
+        with:
+          authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
+          name: ${{ env.CACHIX_NAME }}
+      - name: Add tools needed for non-nix steps
+        run: |  
+          nix-channel --add https://nixos.org/channels/${{ env.NIXPKGS_CHANNEL }} nixpkgs
+          nix-channel --update
+          nix-env -iA nixpkgs.cachix nixpkgs.nodejs nixpkgs.git-lfs nixpkgs.tree nixpkgs.docker
       - uses: actions/checkout@v3
         with:
+          lfs: true
           ref: ${{ github.event.pull_request.head.sha }}
           persist-credentials: false
-      - uses: "./.github/templates/watch-exec"
+      - name: Nix flake show
+        run: |
+          nix --version
+          nix show-config
+          nix flake show --allow-import-from-derivation  --show-trace --fallback --debug --print-build-logs --keep-failed
```

Comment:
Since we are cleaning up; could we lessen the logs necessary? I'm not sure if `ACTIONS_STEP_DEBUG` is accessible in the job, but if that is the case, it would be nice to only add `--show-trace --debug --print-build-logs` conditionally. DEBUG mode can be enabled from the UI (Rerun with debug logs).

Reply:
Tried to do this in 8d92f0c788b8b6a2b9093fdc5188eb2c5448ba9f, but unsure if it works since the runners are down

Reply:
Tried to do this in 8d92f0c788b8b6a2b9093fdc5188eb2c5448ba9f, but unsure if it works since the runners are down

Reply:
I used it like this 
<img width="1840" alt="image" src="https://user-images.githubusercontent.com/757125/216963242-dea94e02-cd6b-404b-b5a6-7a0f48e08de0.png">

Reply:
Ah thanks for the exampke, @dzmitry-lahoda !